internal/impl/aws/input_kinesis.go

```diff
@@ -722,6 +680,67 @@ func (k *kinesisReader) maxShardRunBalancedConsumer(wg *sync.WaitGroup, streamID
 	return numShardClaimed, nil
 }
 
+// Function that attempts to steal shards if it meets the following condition:
+// * It is not its own clientId
+// * It has not reached the max shard number this client can claim
+// * The client it is stealing from has 2 or more shards more than this client currently has
+// It returns True if successful in stealing a shard and any errors if it encountered any
+func (k *kinesisReader) stealShards(wg *sync.WaitGroup, streamID string, clientClaims map[string][]awsKinesisClientClaim) (bool, error){
+	// There were no unclaimed shards, let's look for a shard to steal.
+	selfClaims := len(clientClaims[k.clientID])
+	maxShardNumber, maxShardExists := k.streamMaxShards[streamID]
+	isSuccess:= false
```

Comment:
Maybe it's easier to reuse the `ErrLeaseNotAcquired` error rather than having a `isSuccess` which is only used for test purpose.
So the function will return only an error and then:
- when `isSuccess` is true (we managed to steal a shard) instead of `break` we can `return nil`
- and after the `for` leap we can return `ErrLeaseNotAcquired`
- then the caller can check if the error returned by the function is `ErrLeaseNotAcquired` then it can ignore the error
- We can also update tests to check the returned error type rather than check `isSuccess`

Reply:
Ah okay, thanks :) That would make it slightly cleaner, i was thinking that it is not _technically_ an error so it might add additional confusion. I will put a comment down about it :) I will create a new error because `ErrLeaseNotAcquired` is a very specific error 😅